mrp_bom_attribute_match/tests/test_mrp_bom_attribute_match.py

```diff
@@ -0,0 +1,190 @@
+            sword_cyan = self.sword_attrs.product_template_value_ids[0]
+            with self.assertRaisesRegex(
+                ValidationError,
+                r"You cannot use an attribute value for attribute\(s\) .* in the "
+                r"field “Apply on Variants” as it's the same attribute used in the "
+                r"field “Match on Attribute” related to the component .*",
+            ):
+                line_form.bom_product_template_attribute_value_ids.add(sword_cyan)
+
+    def test_bom_2(self):
+        smell_attribute = self.env["product.attribute"].create(
+            {"name": "Smell", "display_type": "radio", "create_variant": "always"}
+        )
+        orchid_attribute_value_id = self.env["product.attribute.value"].create(
+            [
+                {"name": "Orchid", "attribute_id": smell_attribute.id},
+            ]
+        )
+        plastic_smells_like_orchid = self.env["product.template.attribute.line"].create(
+            {
+                "attribute_id": smell_attribute.id,
+                "product_tmpl_id": self.product_plastic.id,
+                "value_ids": [(4, orchid_attribute_value_id.id)],
+            }
+        )
+        with self.assertRaisesRegex(
+            UserError,
+            r"This product template is used as a component in the BOMs for .* and "
+            r"attribute\(s\) .* is not present in all such product\(s\), and this "
+            r"would break the BOM behavior\.",
+        ):
+            vals = {
+                "attribute_id": smell_attribute.id,
+                "product_tmpl_id": self.product_plastic.id,
+                "value_ids": [(4, orchid_attribute_value_id.id)],
+            }
+            self.product_plastic.write({"attribute_line_ids": [(0, 0, vals)]})
+        mrp_bom_form = Form(self.env["mrp.bom"])
+        mrp_bom_form.product_tmpl_id = self.product_sword
+        with mrp_bom_form.bom_line_ids.new() as line_form:
+            with self.assertRaisesRegex(
+                UserError,
+                r"Some attributes of the dynamic component are not included into "
+                r"production product attributes\.",
+            ):
+                line_form.component_template_id = self.product_plastic
+        plastic_smells_like_orchid.unlink()
+
+    def test_manufacturing_order_1(self):
+        mo_form = Form(self.env["mrp.production"])
+        # added to have product_uom_id.rounding > 0
+        product_uom_id = self.env["uom.uom"].search([])[0]
+        for bom_line in self.bom_id.bom_line_ids:
+            bom_line.product_uom_id = product_uom_id
+        # ---
+        # added to have order uom category as the product uom category
+        mo_form._values[
+            "product_uom_category_id"
+        ] = self.product_sword.uom_id.category_id
+        mo_form._values["product_uom_id"] = self.product_sword.uom_id
+        # --
+        mo_form.product_id = self.product_sword.product_variant_ids.filtered(
```

Comment:
@giarve Hi, as you can see I did a lot of tries to fix test, forcing data to pass to them as first approach.
I have two problem: locally I have a probem with uom in this point, but the strange one is that when I push here I got another one.
I will try to sole the problem but let me know if you see a solutions.

Reply:
@mymage Is the error locally the same I gave in the previous comment?

The problem with the UoM I think appears because of the groups="" attribute on the uom field in the bom line view

If you activate Developer/debug mode and enable the "Manage multiple units of measure" does it still fail?

Reply:
@mymage a good samaritan in discord (OCA server) is suggesting setting the required=True in the view for product_uom_id to force triggering defaults for this field